src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -598,41 +598,41 @@ internal void Cancel(object caller)
                 // Keep looping until we either grabbed the lock (and therefore sent attention) or the connection closes\breaks
                 while ((!hasLock) && (_parser.State != TdsParserState.Closed) && (_parser.State != TdsParserState.Broken))
                 {
+                    // try to take the lock so that if another command is attempted it will queue on the lock
+                    // but don't require that the lock be taken because otherwise attention cannot be sent
+                    // during command execution causing cancellation to wait
+                    // This lock is also protecting against concurrent close and async continuations
                     Monitor.TryEnter(this, _waitForCancellationLockPollTimeout, ref hasLock);
```

Comment:
What's the point of taking a lock at all here? If we actually need to prevent new commands being started while sending attention, then we can't allow this lock to be optional, and if we don't, then why have it at all.